packages/main/src/Calendar.js

```diff
@@ -416,11 +490,83 @@ class Calendar extends UI5Element {
 				this._hideMonthPicker();
 			}
 		}
+
+		if (isTabNext(event)) {
+			this._handleTabNext(event);
+		}
+
+		if (isTabPrevious(event)) {
+			this._handleTabPrevous(event);
+		}
+	}
+
+	_handleTabNext(event) {
+		const target = event.target;
+
+		if (target.tagName === "UI5-DAYPICKER" || target.tagName === "UI5-MONTHPICKER" || target.tagName === "UI5-YEARPICKER") {
+			if (this.monthButton.getAttribute("hidden") === null) {
+				this.monthButton.focus();
+			} else {
+				this.yearButton.focus();
+			}
+			event.preventDefault();
+		} else if (target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Month") {
+			this.yearButton.focus();
+			event.preventDefault();
+		} else {
+			this._setPickerCurrentTabindex(-1);
+		}
+	}
+
+	_handleTabPrevous(event) {
+		const target = event.target;
+
+		if (target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Month") {
+			this._moveFocusToPickerContent();
+			event.preventDefault();
+		} else if (target.tagName === "UI5-CALENDAR-HEADER" && event.path[0].getAttribute("data-sap-show-picker") === "Year") {
+			if (this.monthButton.getAttribute("hidden") === null) {
+				this.monthButton.focus();
+			} else {
+				this._moveFocusToPickerContent();
+			}
+			event.preventDefault();
+		}
+	}
+
+	_moveFocusToPickerContent() {
+		if (!this._oMonth._hidden) {
+			this.dayPicker._itemNav.focusCurrent();
+		} else if (!this._monthPicker._hidden) {
+			this.monthPicker._itemNav.focusCurrent();
+		} else {
+			this.yearPicker._itemNav.focusCurrent();
+		}
+	}
+
+	_onfocusout(event) {
+		this.monthButton.setAttribute("tabindex", "-1");
+		this.yearButton.setAttribute("tabindex", "-1");
+		this._setPickerCurrentTabindex(0);
+	}
+
+	_setPickerCurrentTabindex(index) {
+		if (this.dayPicker) {
+			this.dayPicker._itemNav._getCurrentItem().setAttribute("tabindex", index.toString());
```

Comment:
Create methods in the dayPicker, monthPicker, yearPicker, that updates currentItem tabIndex. Accessing _itemNav does not look nice.

```js
this.dayPicker.setDayCellTabIndex(0)
this.dayPicker.setMonthCellTabIndex(-1)
this.dayPicker.setYearCellTabIndex(0)
```

Then try to use the same pattern of chaning the tabindex, described above (via state and the template)

Reply:
The methods are added into the corresponding picker components. The tab index is still update on after rendering and not through the templates. Seemed complex to make it work through the template as there is an array of items in the grid and only one of them needs changed tabindex value.